src/screens/Onboarding/StepProfile/index.tsx

```diff
@@ -1,55 +1,317 @@
+import {getDataUriSize} from 'lib/media/util'
+import {isNative, isWeb} from 'platform/detection'
 import {
   DescriptionText,
   OnboardingControls,
   TitleText,
 } from '#/screens/Onboarding/Layout'
 import {Context} from '#/screens/Onboarding/state'
-import {atoms as a} from '#/alf'
+import {AvatarCircle} from '#/screens/Onboarding/StepProfile/AvatarCircle'
+import {AvatarCreatorCircle} from '#/screens/Onboarding/StepProfile/AvatarCreatorCircle'
+import {AvatarCreatorItems} from '#/screens/Onboarding/StepProfile/AvatarCreatorItems'
+import {
+  PlaceholderCanvas,
+  PlaceholderCanvasRef,
+} from '#/screens/Onboarding/StepProfile/PlaceholderCanvas'
+import {atoms as a, useBreakpoints, useTheme} from '#/alf'
 import {Button, ButtonIcon, ButtonText} from '#/components/Button'
+import * as Dialog from '#/components/Dialog'
 import {IconCircle} from '#/components/IconCircle'
 import {ChevronRight_Stroke2_Corner0_Rounded as ChevronRight} from '#/components/icons/Chevron'
+import {CircleInfo_Stroke2_Corner0_Rounded} from '#/components/icons/CircleInfo'
 import {StreamingLive_Stroke2_Corner0_Rounded as StreamingLive} from '#/components/icons/StreamingLive'
+import {Text} from '#/components/Typography'
+import {AvatarColor, avatarColors, Emoji, emojiItems} from './types'
+
+export interface Avatar {
+  image?: {
+    path: string
+    mime: string
+    size: number
+    width: number
+    height: number
+  }
+  backgroundColor: AvatarColor
+  placeholder: Emoji
+  useCreatedAvatar: boolean
+}
+
+interface IAvatarContext {
+  avatar: Avatar
+  setAvatar: React.Dispatch<React.SetStateAction<Avatar>>
+}
+
+const AvatarContext = React.createContext<IAvatarContext>({} as IAvatarContext)
+export const useAvatar = () => React.useContext(AvatarContext)
+
+const randomColor =
+  avatarColors[Math.floor(Math.random() * avatarColors.length)]
 
 export function StepProfile() {
   const {_} = useLingui()
-  const {dispatch} = React.useContext(Context)
+  const t = useTheme()
+  const {gtMobile} = useBreakpoints()
+  const {track} = useAnalytics()
+  const {requestPhotoAccessIfNeeded} = usePhotoLibraryPermission()
+  const creatorControl = Dialog.useDialogControl()
+  const [error, setError] = React.useState('')
+
+  const {state, dispatch} = React.useContext(Context)
+  const [avatar, setAvatar] = React.useState<Avatar>({
+    image: state.profileStepResults?.image,
+    placeholder: emojiItems.at,
+    backgroundColor: randomColor,
+    useCreatedAvatar: false,
+  })
+
+  const canvasRef = React.useRef<PlaceholderCanvasRef>(null)
+
+  React.useEffect(() => {
+    track('OnboardingV2:StepProfile:Start')
+  }, [track])
+
+  const openPicker = React.useCallback(
+    async (opts?: ImagePickerOptions) => {
+      const response = await launchImageLibraryAsync({
+        exif: false,
+        mediaTypes: MediaTypeOptions.Images,
+        quality: 1,
+        ...opts,
+      })
+
+      return (response.assets ?? [])
+        .slice(0, 1)
```

Comment:
Definitely fine to leave the slice here (Android is weird and _should_ respect the limit but every platform version has a different image picker implementation), but just want to note to anyone else reviewing that the picker only allows one image to be picked.